### ISSUE TITLE
Check that asynchronous receival of logs is fine

### ISSUE THREAD

**Issue by j4mesholland** [ocadotechnology/aimmo#631]

**Comment by OlafSzmidt** [comment]
We need to check if data that comes in from each worker pod (at different times over a connection, hence asynchronous) can be received fine on our game pod receiving end to be gathered into one data structure.

**Comment by riaJha97** [comment]
So, the ConcurrentTurnManager:
 - creates a thread for every active avatar and runs them in parallel
 - Each thread calls register_action(), which internally calls decide_action()
 - The turn manager can only terminate after all the threads that were created have been terminated (using thread.join())
 - The world map is then updated with the new actions, etc. 
The use of thread.join() ensures that this shouldn't really be an issue

**Comment by mrniket** [comment]
Does the thread also fetch the action from the game-workers?

**Comment by OlafSzmidt** [comment]
Looks good! Means we don't have to worry too much as it will always be gathered at the same time. It's kind of a "all or nothing" policy.

But this makes me think, `decide_action` will make the `_fetch-action` call. Can `requests.post` block one of the threads at the time of failure?

**Comment by riaJha97** [comment]
I was also wondering if the number of avatars could be an issue at some point? Since we create a thread for each avatar, we might exceed the number of HTTPConnections when sending the post request or something.

**Comment by mrniket** [comment]
Worth checking, but I would say that's ⚠️ out of scope for this issue

**Comment by OlafSzmidt** [comment]
But needs documenting in an issue though
